### PR TITLE
Strict MVars without invariant checking

### PR DESCRIPTION
**Pull request by jorisdral**
This PR provides a new `strict-mvar` packages that implements strict MVars without invariant checking. Strict MVars are implemented in a fairly straightforward way by wrapping lazy MVars and forcing evaluation to WHNF before placing values in the lazy MVar.

@dcoutts suggested that strict MVars with invariant checking should be provided separately from MVars without invariant checking. As such, this PR only provides the latter.

### CODE REVIEW

File: strict-mvar/strict-mvar.cabal [hunk 1]
```diff
@@ -0,0 +1,43 @@
+cabal-version:       2.0
+name:                strict-mvar
+version:             0.1.0.0
+synopsis:            Strict MVars for `IO` and `io-sim`
+description:         Strict MVars for `IO` and `io-sim`.
```
Comment (jorisdral) [inline strict-mvar/strict-mvar.cabal:5]: This should probably say `io-classes`

File: strict-mvar/src/Control/Concurrent/Class/MonadMVar/Strict.hs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+modifyMVarMasked :: MonadMVar m => StrictMVar m a -> (a -> m (a,b)) -> m b
+modifyMVarMasked v io = Lazy.modifyMVarMasked (mvar v) io'
+  where
+    io' a = do
+      (!a', b) <- io a
+      pure (a', b)
+
+tryReadMVar :: MonadMVar m => StrictMVar m a -> m (Maybe a)
+tryReadMVar = error "tryReadMVar: no lazy implementation exists"
```
Comment (jorisdral) [inline strict-mvar/src/Control/Concurrent/Class/MonadMVar/Strict.hs:132]: Can be implemented after #70

File: strict-mvar/strict-mvar.cabal [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+copyright:           2019-2023 Input Output Global Inc (IOG).
+author:              IOHK Engineering Team
+maintainer:          operations@iohk.io
+category:            Control
+build-type:          Simple
+tested-with:         GHC == 8.10.7, GHC == 9.2.5, GHC == 9.4.3
+
+source-repository head
+  type:     git
+  location: https://github.com/input-output-hk/ouroboros-network
```
Comment (jorisdral) [inline strict-mvar/strict-mvar.cabal]: This is `s-r-p` is wrong

File: strict-mvar/strict-mvar.cabal [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  manual:      False
+  default:     False
+
+library
+  hs-source-dirs:      src
+
+  exposed-modules:     Control.Concurrent.Class.MonadMVar.Strict
+  default-language:    Haskell2010
+  build-depends:       base        >= 4.9 && <4.18,
+                       io-classes  >= 0.3 && <0.5
```
Comment (coot) [inline strict-mvar/strict-mvar.cabal:33]: we're about to publish `0.5`, so
```suggestion
                       io-classes  >= 0.3 && <0.6
```

**Review by coot — CHANGES_REQUESTED** [review]
Very nice, just a few minor remarks.

File: strict-mvar/strict-mvar.cabal [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+license:             Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+copyright:           2019-2023 Input Output Global Inc (IOG).
+author:              IOHK Engineering Team
+maintainer:          operations@iohk.io
+category:            Control
+build-type:          Simple
+tested-with:         GHC == 8.10.7, GHC == 9.2.5, GHC == 9.4.3
```
[the remark this answers is not in the document]
Comment (jorisdral) [reply strict-mvar/strict-mvar.cabal:15]: Done. Also, `9.4.3` should have been `9.4.4`, so I've changed that

File: strict-mvar/strict-mvar.cabal (hunk 1 above)
Comment (jorisdral) [reply strict-mvar/strict-mvar.cabal:5]: I've changed it to:

> Strict MVars for implementations of the `io-classes` MVar interface

**Pull request merged**